liquidator/liquidator.js

```diff
@@ -249,50 +269,129 @@ class Liquidator {
+            amountWei: tokensToLiquidate.toString()
+          }),
+          this.oneInch.getExpectedReturn({
+            fromToken: tokenCurrency.address,
+            toToken: this.reserveCurrencyAddress,
+            amountWei: this.toWei("1", "gwei")
+          })
         ]);
-      } catch (error) {
-        this.logger.error({
+
+        this.logger.info({
           at: "Liquidator",
-          message:
-            "Failed to liquidate position: not enough synthetic (or large enough approval) to initiate liquidation❌",
-          sponsor: position.sponsor,
-          inputPrice: scaledPrice.toString(),
-          position: position,
-          minLiquidationPrice: this.liquidationMinPrice,
-          maxLiquidationPrice: maxCollateralPerToken.toString(),
-          tokensToLiquidate: tokensToLiquidate.toString(),
-          error
+          message: `Attempting to convert reserve currency ${this.reserveCurrencyAddress} to tokenCurrency ${tokenCurrency.address} ❌`
         });
-        continue;
-      }
 
-      const txnConfig = {
-        from: this.account,
-        gas: Math.min(Math.floor(gasEstimation * this.GAS_LIMIT_BUFFER), this.txnGasLimit),
-        gasPrice: this.gasEstimator.getCurrentFastPrice()
-      };
-      this.logger.debug({
-        at: "Liquidator",
-        message: "Liquidating position",
-        position: position,
-        inputPrice: scaledPrice.toString(),
-        minLiquidationPrice: this.liquidationMinPrice,
-        maxLiquidationPrice: maxCollateralPerToken.toString(),
-        tokensToLiquidate: tokensToLiquidate.toString(),
-        liquidationId: liquidationResult.liquidationId,
-        tokensLiquidated: liquidationResult.tokensLiquidated.rawValue.toString(),
-        finalFeeBond: liquidationResult.finalFeeBond.rawValue.toString(),
-        txnConfig
-      });
+        const reserveWeiNeededBN = this.toBN(reserveWeiNeeded);
+        const oneGweiReturnBN = this.toBN(oneGweiReturn);
+        const tokensToLiquidateBN = this.toBN(tokensToLiquidate);
+        const oneGweiBN = this.toBN(this.toWei("1", "gwei"));
+        const oneWeiBN = this.toBN(this.toWei("1"));
+
+        // Slippage = (idealReserveWeiNeeded - reserveWeiNeeded) / idealReserveWeiNeeded
+        // Where idealReserverWei = oneGweiReturn * tokensToLiquidate / 1 gwei
+        const idealReserveWeiNeededBN = oneGweiReturnBN.mul(tokensToLiquidateBN).div(oneGweiBN);
+
+        let slippage;
+        // IdealReserveWeiNeededBN is 0 if it isn't on exchanges yet
+        // Can't div by 0 so doing this
+        if (idealReserveWeiNeededBN.eq(this.toBN("0"))) {
+          slippage = "0";
+        } else {
+          slippage = this.fromWei(
+            idealReserveWeiNeededBN
+              .sub(reserveWeiNeededBN)
+              .mul(oneWeiBN)
+              .div(idealReserveWeiNeededBN)
+          );
+        }
+
+        // Don't allow slippage of more than 10%
+        if (parseFloat(slippage) > 0.1) {
+          this.logger.debug({
```

Comment:
should this prehaps be a `info` level event? I think we'd like to know about this if this was to happen.